islandora_batch.install

```diff
@@ -45,7 +45,7 @@ function islandora_batch_schema() {
         'not null' => FALSE,
       ),
     ),
-    'primary key' => array('id'),
+    'primary key' => array('id', 'sid'),
```

Comment:
Not... sure this is valid, given all the joins which happen... everything thing that refers to this as a foreign key would be broken, such as the state, resource and message tracking (the `islandora_batch_state`, `islandora_batch_resources` and `islandora_batch_queue_messages`, respectively) are all associated with the unique entry in the queue... it would now be sharing things across sets, which probably isn't desirable... `state` especially... and might be desirable to reword some of the things: instead of "per-object" messages, "per-queue entry", kind of thing...

Reply:
"broken" might be the wrong word... more like... consistent in unexpected ways, when navigating the GUI?

Reply:
Just to add to the Foreign key conversation https://interworks.com/blog/jkhalaj/2012/04/12/drupal-7-usage-foreign-keys-schema-api-and-current-default-fk-erd/
and official statement https://www.drupal.org/node/224333#foreign-keys-added

Reply:
At a glance, the gist looks fine (a diff/something highlighting the changes would be more readable), but it's not so much the schema itself I'm concerned with, 'cause yeah, foreign keys are just descriptive as you point out; however, they're nice to keep up-to-date for to facilitate these types of discussions...

What I'm more concerned with is more the things like the various state checks:
* https://github.com/Islandora/islandora_batch/blob/c4f31b79fe4132c46f20472f8a88ba3dd994bbf3/includes/ingest.batch.inc#L295-L296
* https://github.com/Islandora/islandora_batch/blob/c4f31b79fe4132c46f20472f8a88ba3dd994bbf3/includes/ingest.batch.inc#L307-L309
* https://github.com/Islandora/islandora_batch/blob/c4f31b79fe4132c46f20472f8a88ba3dd994bbf3/includes/ingest.batch.inc#L329-L330

... The checks on parents as well would likely have to change, since they can no longer be the single columns check... probably safe to constrain to the same set: https://github.com/Islandora/islandora_batch/blob/c4f31b79fe4132c46f20472f8a88ba3dd994bbf3/includes/ingest.batch.inc#L293

... There would be some views adjustments also in order, to constrain things to items in the given set:
* https://github.com/Islandora/islandora_batch/blob/c4f31b79fe4132c46f20472f8a88ba3dd994bbf3/views/islandora_batch.views.inc#L22-L25
* https://github.com/Islandora/islandora_batch/blob/c4f31b79fe4132c46f20472f8a88ba3dd994bbf3/views/handlers/IslandoraBatchViewsFieldHandlerAction.inc#L134-L153

... along with adjusting the menu routes used by views to accept the set:
* https://github.com/Islandora/islandora_batch/blob/c4f31b79fe4132c46f20472f8a88ba3dd994bbf3/islandora_batch.module#L34-L52

Reply:
Totally. I agree with you @adam-vessey . Those changes are the ones i was speaking about to make this pull not only something that does not break existing functionality and allows other cases, but be fully robust on those future cases. If you agree with the general structure of that gist, i can take that and move forward with changing every place there the assumption is a PID is the only key to rule them all to PID, SET based display and functionality. Will see what my agenda allows me this week, but should be pretty trivial, have already annotated all my findings + yours = feel all based would be covered. Thanks!

Reply:
Ok, coming back to this, sorry for the long wait. Will have an update this week.